### ISSUE TITLE
Properties accessed inside method signature syntax are marked as "never read" by TS6133

### ISSUE THREAD

**Issue by n0nick** [Microsoft/TypeScript#20769]
**TypeScript Version:**  2.7.0-dev.20171216

**Code**

```ts
class A {
  private foo;
  private bar;

  func() {
    console.log({
      a: () => { console.log(this.foo); }, // this is fine
      b() { console.log(this.bar); }, // here's the bug
    });
  }
}
```

**Config**

```json
{
  "include": ["*.ts"],
  "compilerOptions": {
    "noUnusedLocals": true,
    "target": "es5"
  }
}
```

**Expected behavior:**
Both `foo` and `bar` should be marked as "read" (since they're being used by `func`) and so, TS6133 shouldn't complain.

**Actual behavior:**
Since `bar` is used within a "method signature syntax" (`b() { .. }`), its body is not evaluated and `tsc` complains:
```
index.ts(3,11): error TS6133: 'bar' is declared but its value is never read.
```

**Comment by ajafff** [comment]
`this` inside method `b` is not the class instance, but the object passed to `console.log`. The error is correct.

**Comment by n0nick** [comment]
that... is correct :) thanks for catching this.
